src/github.com/getlantern/autoupdate-server/server/server.go

```diff
@@ -108,16 +108,31 @@ func (g *ReleaseManager) CheckForUpdate(p *Params) (res *Result, err error) {
 		return nil, fmt.Errorf("Arch is required")
 	}
 
-	// Looking if there is a newer version for the os/arch.
 	var update *Asset
-	if update, err = g.getProductUpdate(p.OS, p.Arch); err != nil {
-		return nil, fmt.Errorf("Could not lookup for updates: %s", err)
+
+	// This is a hack that allows Lantern 2.0.0beta8+manoto clients to upgrade to
+	// Lantern 2.0.0 using an interim 2.0.0-beta9+manoto version.
+	//
+	// See https://github.com/getlantern/lantern/issues/2868
+	if appVersion.String() == manotoBeta8 {
```

Comment:
Here's a crazy idea: when we get update requests from 2.0.0-beta8 just randomly choose between the two updates to send. If it's the wrong one, theoretically it should fail on the client side, and will just try again later. I'm not actually sure that's true, but it could work if so.